### PR TITLE
ComputePathToPose Sets empty path on blackboard if action is aborted or cancelled

### PR DESCRIPTION
**Pull request by jwallace42**
---

## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | #3090 |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on | Gazebo |

---

## Description of contribution in a few bullet points
This pr sets the path to be empty on the blackboard if compute pose to path fails. It addresses the issue with the behavior tree navigating along a old path using https://github.com/ros-planning/navigation2/blob/main/nav2_bt_navigator/behavior_trees/navigate_w_recovery_and_replanning_only_if_path_becomes_invalid.xml. 

The new behavior will be
    Tries to compute path, fails clears costmap
    Recovery node ticks again, compute path does get ticked again because isPathValid == FAILURE due to the path being empty.

## Description of documentation updates required from your changes

---

#### For Maintainers: 
- [ ] Check that any new parameters added are updated in navigation.ros.org
- [ ] Check that any significant change is added to the migration guide
- [ ] Check that any new features **OR** changes to existing behaviors are reflected in the tuning guide
- [ ] Check that any new functions have Doxygen added
- [ ] Check that any new features have test coverage
- [ ] Check that any new plugins is added to the plugins page
- [ ] If BT Node, Additionally: add to BT's XML index of nodes for groot, BT package's readme table, and BT library lists

### CODE REVIEW

**Comment by jwallace42** [comment]
@automech-rb this should fix the issue you posted. If you get a chance to run it that would be great :).

**Review by SteveMacenski — APPROVED** [review]
Oh good idea, can you also do the `ComputePathThroughPoses` too?

File: nav2_behavior_tree/plugins/action/compute_path_to_pose_action.cpp [hunk 1]
```diff
@@ -43,6 +43,20 @@ BT::NodeStatus ComputePathToPoseAction::on_success()
   return BT::NodeStatus::SUCCESS;
 }
 
+BT::NodeStatus ComputePathToPoseAction::on_aborted()
+{
+  geometry_msgs::msg::PoseStamped empty_path;
```
Comment (SteveMacenski) [inline nav2_behavior_tree/plugins/action/compute_path_to_pose_action.cpp:48]: Wrong type :wink: This should be a Path not a pose stamped. 

https://app.circleci.com/pipelines/github/ros-planning/navigation2/7718/workflows/ff4e7675-61ac-482f-baad-bc0d3fde29ec/jobs/26788

CI failed on this (also @facontidavide great user protective error! That saved us a bunch of time and energy)

**Review by SteveMacenski — CHANGES_REQUESTED** [review]

File: nav2_behavior_tree/plugins/action/compute_path_to_pose_action.cpp (hunk 1 above)
Comment (facontidavide) [reply nav2_behavior_tree/plugins/action/compute_path_to_pose_action.cpp:48]: you are welcome :)

**Review by SteveMacenski — APPROVED** [review]

**Pull request merged**